semgrep/semgrep/test.py

```diff
@@ -176,24 +185,25 @@ def score_output_json(
                 if not has_parseable_rule_id:
                     continue
                 rule_ids = normalize_rule_ids(line)
-                if (not ignore_todo and todo_rule_in_line) or rule_in_line:
-                    for rule_id in rule_ids:
-                        ruleid_lines[test_file_resolved][rule_id].append(
-                            effective_line_num
-                        )
-                if (not ignore_todo and todo_rule_in_line) or ok_in_line:
-                    for rule_id in rule_ids:
-                        ok_lines[test_file_resolved][rule_id].append(effective_line_num)
-                if ignore_todo and todo_ok_in_line:
-                    for rule_id in rule_ids:
-                        todo_ok_lines[test_file_resolved][rule_id].append(
-                            effective_line_num
-                        )
+                if todo_rule_in_line or rule_in_line:
```

Comment:
Trying to understand this behavior here: Does this mean that `todo` lines will still be considered as part of the final set of rules? Will there be any errors if `todo` lines match? Don't match?

Reply:
Ah, if I read the thread correctly this will get removed at a later time

Reply:
yep! sorry for the late reply -- there's no errors if `todo` lines match currently, and the `todo` lines aren't considered as the final set of rules (they're later removed from the reported and expected sets of match line numbers). So currently this is still completely functional 👍